File: tests/e2e/Dockerfile.go
FROM golang:1.21-alpine

WORKDIR /app
COPY sdks/go/go.mod .
COPY sdks/go/thea/ thea/

# Create a cmd directory for the e2e test
RUN mkdir -p cmd/e2e
COPY tests/e2e/test_e2e.go cmd/e2e/main.go

RUN cd cmd/e2e && go build -o /test_e2e .
CMD ["/test_e2e"]

File: tests/e2e/Dockerfile.java
FROM maven:3.9-eclipse-temurin-17 AS build

WORKDIR /sdk
COPY sdks/java/ .
RUN mvn -q package -DskipTests

WORKDIR /app
COPY tests/e2e/TestE2E.java .
RUN javac -cp /sdk/target/thea-recorder-1.0.0.jar TestE2E.java

FROM eclipse-temurin:17-jre
COPY --from=build /sdk/target/thea-recorder-1.0.0.jar /app/
COPY --from=build /app/TestE2E.class /app/
WORKDIR /app
CMD ["java", "-cp", ".:thea-recorder-1.0.0.jar", "TestE2E"]
